src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UseStartsWithInsteadOfIndexOfComparisonWithZeroTests.cs

```diff
@@ -0,0 +1,262 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.UseStartsWithInsteadOfIndexOfComparisonWithZero,
+    Microsoft.NetCore.Analyzers.Performance.UseStartsWithInsteadOfIndexOfComparisonWithZeroCodeFix>;
+
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Performance.UseStartsWithInsteadOfIndexOfComparisonWithZero,
+    Microsoft.NetCore.Analyzers.Performance.UseStartsWithInsteadOfIndexOfComparisonWithZeroCodeFix>;
+
+namespace Microsoft.NetCore.Analyzers.Performance.UnitTests
+{
+    public class UseStartsWithInsteadOfIndexOfComparisonWithZeroTests
```

Comment:
>UseStartsWithInsteadOfIndexOfComparisonWithZeroTests [](http://example.com/codeflow?start=17&length=52)

We need to add a test cases for IndexOf overloads which takes 2 arguments (with StringComparison) too.

Reply:
@tarekgh What about `IndexOf(char, StringComparison)`? There is no char overload in `StartsWith` that accepts StringComparison. Should we not report? Or use `StartsWith(string, StringComparison)`? Or something else?

Reply:
That is a good question. If it will not be complicated to implement, we can use [StartsWith(ReadOnlySpan<Char>, ReadOnlySpan<Char>, StringComparison)](https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.startswith?view=net-7.0#system-memoryextensions-startswith(system-readonlyspan((system-char))-system-readonlyspan((system-char))-system-stringcomparison)). In general, I see the `IndexOf` overloads that takes string is higher priority as these used in main scenario. We can support the `char` overloads later if we need to.

Reply:
@stephentoub What do you think?

Reply:
@Youssef1313 how complex would it be if implement `IndexOf(char, StringComparison)` as I suggested above?

Reply:
Found https://github.com/dotnet/roslyn/issues/61432, but it doesn't seem like the comment there applies in the test project in this repo.

@sharwell Can you help please?

Reply:
No, the whole point of this analyzer is the performance. 

for `s.IndexOf(c) == 0` you can convert it to 

```C#
s.length > 0 && s[0] == c
```

Reply:
Makes sense.

I started looking into `IndexOf(char, StringComparison)` and there is one thing.

> we can use [StartsWith(ReadOnlySpan, ReadOnlySpan, StringComparison)](https://learn.microsoft.com/en-us/dotnet/api/system.memoryextensions.startswith?view=net-7.0#system-memoryextensions-startswith(system-readonlyspan((system-char))-system-readonlyspan((system-char))-system-stringcomparison))

Here, we need to change the `char` to a `ReadOnlySpan<char>`

The `ReadOnlySpan<T>(T)` ctor is only available in .NET 7, the other way is to use `stackalloc char[1] { ch }`. Is that okay? What about VB? I'm also not sure about `ReadOnlySpan` (`ref struct`s in general) support in VB.

Reply:
> to use stackalloc char[1] { ch }. Is that okay?

Yes, this should be Ok.  You just convert the IndexOf call too `string.AsSpan().StartsWith(stackalloc char[1] { chr }, stringComparison);`

> What about VB? I'm also not sure about ReadOnlySpan (ref structs in general) support in VB.

Considering it is rare to find the pattern `IndexOf(char, StringComparison) == 0` in VB. I think for this case we either exclude this rule from there or we can go with the fixer that allocates the string from the character. I mean use `StartsWith(char.ToString(), , StringComparison)`.